iceoryx_hoofs/container/include/iox/detail/vector.inl

```diff
@@ -391,9 +447,16 @@ inline const T& vector<T, Capacity>::at_unchecked(const uint64_t index) const no
 template <typename T, uint64_t Capacity>
 inline void vector<T, Capacity>::clearFrom(const uint64_t startPosition) noexcept
 {
-    while (m_size > startPosition)
+    if constexpr (std::is_trivial<T>::value && std::is_class<T>::value)
```

Comment:
The conditions when to call the dtor should be the same like in `erase`

Reply:
done!